os/kernel/log_dump/log_dump.c

```diff
@@ -505,6 +515,17 @@ int log_dump_save(char ch)
 	return LOG_DUMP_OK;
 }
 
+size_t log_dump_get_size(void)
+{
+	/* If log_dump save is stopped, that means partially uncompressed buffer
+	 * is compressed and size should include last_comp_block_size */
+	if (is_started_to_save == false) {
+		return log_dump_chunk_node_cnt * CONFIG_LOG_DUMP_CHUNK_SIZE + compress_curbytes + last_comp_block_size;
+	} else {
```

Comment:
The else case means, top_logsave has not been called. So, if we return some size here and there is some timegap before read, then the size will differ. It is better to return error code here. This api must be called only when log dump has been stopped.

Reply:
done

Reply:
Here you added the else case to check stop first. How about adding it at the read function as well? Because it's the same.